hive-metastore/src/main/java/org/apache/iceberg/hive/HiveTableOperations.java

```diff
@@ -263,6 +266,15 @@ protected void doCommit(TableMetadata base, TableMetadata metadata) {
         throw e;
 
       } catch (Throwable e) {
+        if (e.getMessage()
```

Comment:
This is a bit unfortunate, we didn't go with a proper subclass of MetaException

Reply:
IIRC Thrift does not work well with subclassing

Reply:
Hm. how about NoSuchObjectException ?  That seems to propagate to Iceberg code somehow?

Reply:
That is specifically set as a possible returned exception already on the API. About 2 years ago I have proposed to clean up the exception handling by adding/removing exceptions to the methods, but my suggestion got rejected as it would be a breaking change.